### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by based-kwl**
Retrieve user information via a get request

### CODE REVIEW

File: src/routes/users.js [hunk 1]
```diff
@@ -19,6 +19,24 @@ router.route('/:id').get((req, res) => {
...
+    User.findById(req.params.username)
+        .then(user => res.json(user))
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Get user by email
+ */
+ router.route('/:email').get((req, res) => {
+    User.findById(req.params.email)
```
Comment (guaj) [inline src/routes/users.js:35]: Email is not the id. Username is. using the email in User.findById will always return null. check login route to check how to find a user by email.

File: src/routes/users.js [hunk 2]
```diff
@@ -19,6 +19,24 @@ router.route('/:id').get((req, res) => {
         .catch(err => res.status(400).json('Error: ' + err));
 });
 
+/**
+ * Get user by username
+ */
+ router.route('/:username').get((req, res) => {
+    User.findById(req.params.username)
```
Comment (guaj) [inline src/routes/users.js:26]: Have you tested this? Username is the id in our ER diagram, but I don't see anywhere in mongoose where you linked username to _id of the model. Please refer to login route to search a user by attribute or specify in the model that username is the id.

**Review by guaj — CHANGES_REQUESTED** [review]
You need to find a way to test your code before doing a PR. There are multiple ways to test (Postman, Jest, etc.). This code is not working.

**Comment by based-kwl** [comment]
Rewrote the routes and tested them with postman this time (sorry about that, I just assumed it worked the same way since they were unique keys).

**Review by guaj — APPROVED** [review]

**Pull request merged**